### PR TITLE
Build script for Python wheels for OS/X

### PR DESCRIPTION
**Pull request by hernando**

### CODE REVIEW

File: brain/python/CMakeLists.txt [hunk 1]
```diff
@@ -58,6 +58,10 @@ set(_brain_python_libs Brain ${Boost_PYTHON${USE_BOOST_PYTHON_VERSION}_LIBRARY})
 # link python libraries.
 if(COMMON_LIBRARY_TYPE MATCHES "SHARED")
   list(APPEND _brain_python_libs ${PYTHON_LIBRARIES})
+else()
+  if(APPLE)
+    set_target_properties(brain_python PROPERTIES LINK_FLAGS "-undefined dynamic_lookup")
```
Comment (tribal-tec) [inline brain/python/CMakeLists.txt]: comment what this does?

File: packaging/python_wheels/mac/build.sh [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+
+WORKSPACE=$(realpath `pwd`/../../../..)
+SOURCE_PATH=$WORKSPACE/Brion
+
+# Cannot use an earlier version because C++11 features don't compile otherwise
+export MACOSX_DEPLOYMENT_TARGET="10.9"
+export PYTHON_CONFIGURE_OPTS="--enable-universalsdk=/ --with-universal-archs=intel"
+export PYENV_ROOT=$WORKSPACE/pyenv
+export PATH="$PYENV_ROOT/bin:$PATH"
+export HTTP_PROXY=bbpproxy.epfl.ch:80
```
Comment (tribal-tec) [inline packaging/python_wheels/mac/build.sh]: this is already setup'd by CI, really needed here?

File: packaging/python_wheels/mac/build.sh [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+export HTTPS_PROXY=bbpproxy.epfl.ch:80
+export http_proxy=bbpproxy.epfl.ch:80
+export https_proxy=bbpproxy.epfl.ch:80
+
+NUMPY_VERSION=1.12.0
```
Comment (tribal-tec) [inline packaging/python_wheels/mac/build.sh]: I assume those versions are they same as for the linux build. Can we unify them in a common place?

File: packaging/python_wheels/mac/build.sh [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+      -DCLONE_SUBPROJECTS=ON                                          \
+      -DUSE_PYTHON_VERSION=$(get_python_major_version)                \
+      -DBoost_USE_STATIC_LIBS=ON                                      \
+      -DHDF5_USE_STATIC_LIBRARIES=ON                                  \
+      -DPYTHON_EXECUTABLE:FILEPATH=$(which python)                    \
+      -DPYTHON_LIBRARY=$(get_python_library)                          \
+      -DPYTHON_INCLUDE_DIR=$(get_python_include)                      \
+      -DCOMMON_LIBRARY_TYPE=STATIC || { popd; return 1; }
+
+    VERBOSE=1 make -j2 brain_python || { popd; return 1; }
```
Comment (tribal-tec) [inline packaging/python_wheels/mac/build.sh]: really verbose?

File: packaging/python_wheels/mac/build.sh [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+export PATH="$PYENV_ROOT/bin:$PATH"
+export HTTP_PROXY=bbpproxy.epfl.ch:80
+export HTTPS_PROXY=bbpproxy.epfl.ch:80
+export http_proxy=bbpproxy.epfl.ch:80
+export https_proxy=bbpproxy.epfl.ch:80
+
+NUMPY_VERSION=1.12.0
+SPHINX_VERSION=1.3.6
+PYTHON_VERSIONS="2.7.14 3.4.7 3.5.4 3.6.4"
+
```
Comment (tribal-tec) [inline packaging/python_wheels/mac/build.sh]: all the other versions for boost, hdf5, etc. should also be grouped. and ideally shared with the linux build instructions.

File: packaging/python_wheels/mac/build.sh [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+{
+    create_venv test
+    . test/bin/activate
+
+    pip install brain*.whl
+    python -c "import brain; brain.Circuit;" || { deactivate; return 1; }
+
+    deactivate
+
+    # Once tested, the whell is moved to wheelhouse
```
Comment (tribal-tec) [inline packaging/python_wheels/mac/build.sh]: typo 'whell'

File: packaging/python_wheels/mac/build.sh (hunk 2 above)
Comment (hernando) [reply packaging/python_wheels/mac/build.sh]: If my memory doesn't fail, I remember having network errors, which is why I added them. The build_wheel.sh script for Linux does this:

    export http_proxy=${HTTP_PROXY-$http_proxy}
    export https_proxy=${HTTPS_PROXY-$https_proxy}

So I will replace these lines with that.

File: packaging/python_wheels/mac/build.sh (hunk 3 above)
Comment (hernando) [reply packaging/python_wheels/mac/build.sh]: will do

File: packaging/python_wheels/mac/build.sh (hunk 5 above)
Comment (hernando) [reply packaging/python_wheels/mac/build.sh]: PYTHON_VERSIONS doesn't seem easy to unify at all. In the Linux script the variable contains a portion of paths inside the Docker container used to locate the different python versions. In fact, it's not really a Python version, but a CPython ABI version.

File: brain/python/CMakeLists.txt (hunk 1 above)
Comment (hernando) [reply brain/python/CMakeLists.txt]: Done. By the way, I also had to add an equivalent to --whole-archive

**Pull request merged**